packages/event-handler/src/rest/RouteHandlerRegistry.ts

```diff
@@ -13,6 +27,56 @@ class RouteHandlerRegistry {
     this.#logger = options.logger;
   }
 
+  /**
+   * Compares two dynamic routes to determine their specificity order.
+   * Routes with fewer parameters and more path segments are considered more specific.
+   * @param a - First dynamic route to compare
+   * @param b - Second dynamic route to compare
+   * @returns Negative if a is more specific, positive if b is more specific, 0 if equal
+   */
+  #compareRouteSpecificity(a: DynamicRoute, b: DynamicRoute): number {
+    // Routes with fewer parameters are more specific
+    const aParams = a.paramNames.length;
+    const bParams = b.paramNames.length;
+
+    if (aParams !== bParams) {
+      return aParams - bParams;
+    }
+
+    // Routes with more path segments are more specific
+    const aSegments = a.path.split('/').length;
+    const bSegments = b.path.split('/').length;
+
+    return bSegments - aSegments;
+  }
+  /**
+   * Adds a dynamic route to the registry, ensuring that the dynamic
+   * routes array says sorted in ascending order of specificity.
+   * @param route - The dynamic route to add
+   * @returns The index in the specific routes array where the route was stored
+   */
+  #addDynamicRoute(route: DynamicRoute): number {
+    for (let i = 0; i < this.#dynamicRoutes.length; i++) {
+      const existingRoute = this.#dynamicRoutes[i];
+      if (this.#compareRouteSpecificity(route, existingRoute) < 0) {
+        this.#dynamicRoutes.splice(i, 0, route);
```

Comment:
Note, the time complexity of `splice` is basically O(n) when the item to be inserted is towards the end of the list. I had thought of using a priority heap instead of an array, which is O(log n) for these sorts of operations, but thought that using a non-standard data structure that we'd have to implement ourselves might be overkill. It is very unlikely that users will ever register enough routes for this to become a problem.

Reply:
Ignore this, I realised we can just use a simple binary search and get O(n) performance.

Reply:
> I've realised that rather than doing the sort eagerly, we can do it lazily instead by setting a flag to indicate whether the list should be sorted that the resolve method checks. As resolve will only ever be called after all the routes have been registered, the call to sort will only ever be done once.

Smart - well done with this analysis.